wikimedia/wikipedia-ios
# Multilingual Campaign Support

**Phabricator:** 
https://phabricator.wikimedia.org/T347352
https://phabricator.wikimedia.org/T345851

### Notes
This PR adds the necessary calls in `WMFAnnouncementsContentSource.m` to fetch our new campaign config ([prod](https://donate.wikimedia.org/wiki/MediaWiki:AppsCampaignConfig.json) and [staging](https://test.wikimedia.org/wiki/MediaWiki:AppsCampaignConfig.json)). This data is fetched with the new `WKFundraisingCampaignDataController`, which controls persistent caching of this call as well as handles logic of campaign expirations, modal dismissals, and maybe later.

As a part of this PR I also set up logic for the prompt a user sees after tapping "Donate", which gives the user a choice of Apple Pay or other payment method.

If a user qualifies for the `NL_2023_11` campaign, AND they have what they need to successfully donate via native Apple Pay, they will see this prompt. If they don't have Apple Pay set up to donate natively, they will push to the in-app web view donate form. If they don't even qualify for the `NL_2023_11` campaign (that is, the campaign is out of date or the user's device country code does not return "NL"), they will see the old behavior of kicking them out to Safari.

This PR is dependent on https://github.com/wikimedia/wikipedia-ios/pull/4641

Note: the last handful of commits starting with 2760603 are reacting to changes the FR Tech endpoints need (changing parameters on submitPayment, adding recurring, better error handling, etc.)

### Test Steps
1. Follow steps 1 - 5 on the dependent PR to set the merchant ID and set up your device for sandbox testing.
2. Change your device region to Netherlands in iOS Settings.
3. Launch fresh-installed Staging app. Background, foreground, then pull to refresh on Explore feed to fetch new campaigns and info needed for native donate form.
4. Go to an article on EN or NL Wikipedia. Confirm you see basic alert view with campaign text.
5. Tap "Donate".
6. Confirm you see another alert view asking for Apple Pay vs Other Payment Method. Tap "Apple Pay".
7. Confirm native donate form shows.
8. Back out, confirm going back to article does not show campaign message.
9. Repeat steps 3-4. This time tap "Maybe later". Confirm going back to article does not show campaign text.
10. Change device time to one day later and relaunch. Confirm you see campaign text again.
11. Repeat steps 3-4. This time tap "I already donated". Confirm going back to article does not show campaign text.
12. Repeat steps 3 so that campaign is fetched. Then change device time to late November after the campaign ends. Relaunch app and go to an article. Confirm you do not see campaign message.

Note: Let me know if you need to change the campaign dates here for testing - https://test.wikipedia.org/wiki/MediaWiki:AppsCampaignConfig.json. Alternatively you can locally revert [6eaaa0b] to run the app against mocks and freely edit the campaign dates.
